multimedia/tsduck/Portfile

```diff
@@ -0,0 +1,45 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           makefile 1.0
+PortGroup           github 1.0
+
+github.setup        tsduck tsduck 3.22-1953 v
+
+categories          multimedia
+license             BSD
+maintainers         {mit.edu:quentin @quentinmit} \
+                    openmaintainer
+description         MPEG Transport Stream Toolkit
+long_description    MPEG Transport Stream Toolkit
+
+homepage            https://tsduck.io/
+platforms           darwin
+
+checksums           sha1    cedc70686415a9062bee8c3b1f43af8ea74f2127 \
+                    rmd160  5a7c0b73ab1b6aa9d88b9d6ecde9befb00747f0f \
+                    sha256  ca4e0af170616eb7ea25b3555a2519b94b8443b3135a4681993b5ae54316f5d9 \
+                    size    13567538
+
+depends_build-append bin:gsed:gsed \
+		     bin:dos2unix:dos2unix
```

Comment:
The `bin:` dependency style allows non-MacPorts copies of those executables to satisfy the dependency. We probably don't want that, and should list the dependencies in the normal `port:` style instead.

Reply:
Are these dependencies really needed? For example, dos2unix seems to only be used by the unixify target, but that looks like it's only used by the developer during development. I can't tell whether gsed is used during the build because the build uses silent rules. Please disable silent rules so that we can see the actual build commands.

Reply:
The Makefiles do not support disabling silent rules. I believe the only way to do that would be to depend on `gmake` and run `gmake --trace`, but I don't think that's worth adding a dependency over. LMK if you disagree.

Yes, it uses `gsed` during the build. You're right that it looks like `dos2unix` is only for the `unixify` target which is not invoked by `make install`.

Reply:
> The Makefiles do not support disabling silent rules.

Then we should patch the Makefiles. I did it; it's a fairly small change. Ok if I add it to the Portfile? It has already shown me that there are several more problems to fix:

* flags like `-I/usr/local/include` and `-L/usr/local/lib` and `-lstdc++` should be removed
* `-I/opt/local/include` appears twice: once near the end of the line where it's fine and once near the beginning, before the project's own `-I` flags, which needs to be removed
* the project uses C11 and C++11 which should be indicated with `compiler.c_standard 2011` and `compiler.cxx_standard 2011`
* `-Werror` is causing a build to fail and should be removed

Reply:
> Ok if I add it to the Portfile?

I'll just merge your PR as is for now and I'll send you other PRs for the changes later.